domain/src/main/java/com/example/ScreeningService.java

```diff
@@ -0,0 +1,31 @@
+package com.example;
+
+import com.example.dto.MovieScreeningResponse;
+import com.example.entity.Movie;
+import com.example.entity.Screening;
+import com.example.repository.ScreeningRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+@Service
+@RequiredArgsConstructor
+public class ScreeningService {
+
+    private final ScreeningRepository screeningRepository;
+
+    public List<MovieScreeningResponse> getSchedules(Long theaterId) {
+        List<Screening> screening = screeningRepository.findByTheaterId(theaterId);
+
+        Map<Movie, List<Screening>> groupedByScreen = screening.stream()
+                .collect(Collectors.groupingBy(Screening::getMovie));
+
+        return groupedByScreen.entrySet().stream()
+                .sorted((e1, e2) -> e2.getKey().getReleaseDate().compareTo(e1.getKey().getReleaseDate()))
+                .map(entry -> MovieScreeningResponse.from(entry.getKey(), entry.getValue()))
+                .collect(Collectors.toList());
```

Comment:
이 부분은 스트림을 사용해서 잘 처리해주셨네요! 

다만 한 줄에 너무 많은 내용이 들어가 있어서 가독성이 다소 떨어지는 것 같습니다. 
아래와 같은 형태로 리팩토링 하는건 어떨까요? 

```suggestion
    return groupedByScreen.entrySet().stream()
        // 상영일 기준 내림차순 정렬
        .sorted(Comparator.comparing(entry -> entry.getKey().getReleaseDate(), 
                Comparator.reverseOrder()))
        // 엔트리를 MovieScreeningResponse로 변환
        .map(entry -> MovieScreeningResponse.from(
                entry.getKey(), 
                entry.getValue()))
        .collect(Collectors.toList());
```

Reply:
확실히 가독성이 더 좋네요ㅎㅎ. 수정했습니다!!